### PR TITLE
Create Movie Session HomWork

### PR DESCRIPTION
**Pull request by yuriikozar**
Solution + Checkstyle
v1

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        LocalDateTime max = LocalDateTime.of(date, LocalTime.MAX);
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> findSessionsQuery = session.createQuery("from MovieSession m "
+                    + "WHERE m.id = :id AND m.showTime BETWEEN :min AND :max", MovieSession.class);
+            findSessionsQuery.setParameter("id", movieId);
+            findSessionsQuery.setParameter("min", min);
+            findSessionsQuery.setParameter("max", max);
+            return findSessionsQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find available sessions :(", e);
```
Comment (JadeRowan) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:62]:
```suggestion
            throw new DataProcessingException("Can't find available sessions for date " 
                    + date + ", and movieId " + id, e);
```

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        LocalDateTime min = LocalDateTime.of(date, LocalTime.MIN);
+        LocalDateTime max = LocalDateTime.of(date, LocalTime.MAX);
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> findSessionsQuery = session.createQuery("from MovieSession m "
+                    + "WHERE m.id = :id AND m.showTime BETWEEN :min AND :max", MovieSession.class);
```
Comment (JadeRowan) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:56]: You shoudn't compare `MovieSession.id == Movie.id`, уou have to compare `MovieSession.Movie.id == Movie.id`

File: src/main/java/mate/academy/model/MovieSession.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "movie_id")
+    private Movie movie;
+    @ManyToOne
+    @JoinColumn(name = "cinema_hall_id")
+    private CinemaHall cinemaHall;
+    private LocalDateTime showTime;
```
Comment (JadeRowan) [inline src/main/java/mate/academy/model/MovieSession.java:24]: Use `@Column` annotation to set proper snake_case names for columns

**Review by JadeRowan — APPROVED** [review]

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a cinema hall by id: " + id, e);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:46]: change the message

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**